From 74fc44269613cffef07978b4158c62a9cb887821 Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Fri, 8 Jun 2012 11:03:31 -0400
Subject: [PATCH] slave: only start gnome-session with --debug if debug
 enabled

gnome-session puts all its message in syslog when --debug is
enabled.  This floods the log and makes it hard to see other
messages.

This commit only calls gnome-session with --debug when gdm
debugging is on.

https://bugzilla.gnome.org/show_bug.cgi?id=672801
---
 daemon/gdm-greeter-session.c |   12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/daemon/gdm-greeter-session.c b/daemon/gdm-greeter-session.c
index 44259ab..464b16b 100644
--- a/daemon/gdm-greeter-session.c
+++ b/daemon/gdm-greeter-session.c
@@ -39,6 +39,8 @@
 
 #include "gdm-welcome-session.h"
 #include "gdm-greeter-session.h"
+#include "gdm-settings-direct.h"
+#include "gdm-settings-keys.h"
 
 #define GDM_GREETER_SERVER_DBUS_PATH      "/org/gnome/DisplayManager/GreeterServer"
 #define GDM_GREETER_SERVER_DBUS_INTERFACE "org.gnome.DisplayManager.GreeterServer"
@@ -76,9 +78,17 @@ gdm_greeter_session_new (const char *display_name,
                          gboolean    display_is_local)
 {
         GObject *object;
+        gboolean debug = FALSE;
+        char *command = BINDIR "/gnome-session -f";
+
+        gdm_settings_direct_get_boolean (GDM_KEY_DEBUG, &debug);
+
+        if (debug) {
+                command = BINDIR "/gnome-session -f --debug";
+        }
 
         object = g_object_new (GDM_TYPE_GREETER_SESSION,
-                               "command", BINDIR "/gnome-session -f --debug",
+                               "command", command,
                                "server-dbus-path", GDM_GREETER_SERVER_DBUS_PATH,
                                "server-dbus-interface", GDM_GREETER_SERVER_DBUS_INTERFACE,
                                "server-env-var-name", "GDM_GREETER_DBUS_ADDRESS",
-- 
1.7.10

